scrapers/govScrapers/getSouthAfrica.js

```diff
@@ -0,0 +1,138 @@
+const axios = require('axios');
+const cheerio = require('cheerio');
+const logger = require('../../utils/logger');
+
+/**
+ * Scrapes more granular South African data from Wikipedia (based on Government sources)
+ */
+const southAfricaData = async () => {
+	try {
+		const html = cheerio.load(
+			(
+				await axios.get(
+					'https://en.wikipedia.org/wiki/Template:COVID-19_pandemic_data/South_Africa_medical_cases'
+				)
+			).data
+		);
+
+		var data = {
+			national: {
+				timeline: []
+			},
+			provinces: [
+				{
+					name: 'Eastern Cape',
+					timeline: []
+				},
+				{
+					name: 'Free State',
+					timeline: []
+				},
+				{
+					name: 'Gauteng',
+					timeline: []
+				},
+				{
+					name: 'KwaZulu-Natal',
+					timeline: []
+				},
+				{
+					name: 'Limpopo',
+					timeline: []
+				},
+				{
+					name: 'Mpumalanga',
+					timeline: []
+				},
+				{
+					name: 'Northern Cape',
+					timeline: []
+				},
+				{
+					name: 'North West',
+					timeline: []
+				},
+				{
+					name: 'Western Cape',
+					timeline: []
+				}
+			]
+		};
+
+		var tableRows = html(
+			"span:contains('COVID-19 confirmed cases in South Africa by province')"
+		)
+			.closest('table')
+			.find('tr');
+
+		tableRows.slice(2, tableRows.length - 2).each((i, tr) => {
+			var tds = html(tr).children();
+			var date = `2020-${tds.eq(0).text()}`.replace(/\n/g, '');
+
+			data.national.timeline.push({
+				date: date,
+				cases: {
+					cumulative: parseInt(tds.eq(15).text().replace(/\n|\D/g, '')) || null,
+					unallocated:
+            parseInt(tds.eq(13).text().replace(/\n|\D/g, '')) || null,
+					new: parseInt(tds.eq(14).text().replace(/\n|\D/g, '')) || null
+				},
+				tests: {
+					cumulative: parseInt(tds.eq(3).text().replace(/\n|\D/g, '')) || null
```

Comment:
instead of this line of code so many different times, put it into a function
```js
getCellData(index) {
  return parseInt(tds.eq(index).text().replace(/\n|\D/g, '')) || null
}
```